mesonbuild/mesonmain.py

```diff
@@ -41,6 +41,8 @@ def create_parser():
                    help='Special wrap mode to use')
     p.add_argument('--profile-self', action='store_true', dest='profile',
                    help=argparse.SUPPRESS)
+    p.add_argument('--reconfigure', action='store_true',
+                   help='Reconfigure the project using the same options.')
```

Comment:
This is not very helpful on its own. It should at least mention _why_ you'd want to do this.

Reply:
Changed to:
```
help='Set options and reconfigure the project. Useful when new ' +
                        'options have been added to the project and the default value ' +
                        'is not working.'
```
Is that better?

It's difficult to give a full explanation of the context in a command line help message... Do we have a markdown doc page for command line args to give more details?

The reason is when new option is added, backend will trigger a reconfigure, then if you need to change the default value you'll have to reconfigure a 2nd time. Also it could happen that with the default value the project will fail configuration, in that case you're stuck and have to wipe everything, --reconfigure let you change the default value and reconfigure in one step.